### PR TITLE
Publish summary in PR with GitHub Actions

### PR DESCRIPTION
**Pull request by agneum**
- automatically publish CI Checker summary to a pull request (if exists)
- use composite action to combine steps and have one step to set up the CI Checker action
- share the summary artifact between steps to post comment
- no configuration changes required

Example: 
- https://github.com/postgres-ai/green-zone/runs/7539251035?check_suite_focus=true
- https://github.com/postgres-ai/green-zone/pull/8#issuecomment-1196611447

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -62,7 +34,32 @@ inputs:
...
-  image: 'Dockerfile'
+  using: 'composite'
+  steps:
+    - name: DB migration with DLE
+      uses: agneum/migration-action/core@composite-action
+      with:
+        commands: ${{ inputs.commands }}
+        dbname: ${{ inputs.dbname }}
+        migration_envs: ${{ inputs.migration_envs }}
+        download_artifacts: ${{ inputs.download_artifacts }}
```
Comment (NikolayS) [inline action.yml:49]: @agneum Should we distinguish "safe" (PII-free) and "unsafe" artifacts, for security reasons?

File: core/action.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+name: 'Database Lab: realistic DB testing in CI'
+branding:
+  icon: 'database'  
+  color: 'orange'
+description: 'Check database migrations with the Database Lab Engine'
```
Comment (NikolayS) [inline core/action.yml:5]:
```suggestion
description: 'Test DB migrations using DB branches provided by Database Lab Engine (DLE)'
```

File: core/action.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+inputs:
+  owner:
+    description: 'The owner of the repository to check'
```
Comment (NikolayS) [inline core/action.yml:8]:
```suggestion
    description: 'Repository owner'
```

File: core/action.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    required: true
+    default: ${{ github.repository_owner }}
+  repo:
+    description: 'The repo to check'
```
Comment (NikolayS) [inline core/action.yml:12]:
```suggestion
    description: 'Repository'
```

File: core/action.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    required: true
+    default: ${{ github.event.repository.name }}
+  ref:
+    description: 'The input represents a ref parameter, which can be a SHA, branch, or tag'
```
Comment (NikolayS) [inline core/action.yml:16]:
```suggestion
    description: 'Ref (SHA, branch, or tag) that needs to be tested'
```

File: core/action.yml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    required: true
+    default: ${{ github.ref }}
+  pull_request:
+    description: 'Link to the pull request'
```
Comment (NikolayS) [inline core/action.yml:20]:
```suggestion
    description: 'GitHub PR (URL)'
```

File: core/action.yml [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    required: false
+    default: ${{ github.event.pull_request.html_url }}
+  compare:
+    description: 'The compare link'
```
Comment (NikolayS) [inline core/action.yml:24]:
```suggestion
    description: 'GitHub PR comparison URL'
```

File: core/action.yml [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+  author_name:
+    description: 'Author name'
+    required: false
+    default: ${{ github.event.head_commit.author.name }}
+
+  commands:
+    description: 'List of commands to run DB migrations'
+    required: true
+  dbname:
+    description: 'The database that the workflow is running with'
```
Comment (NikolayS) [inline core/action.yml:40]:
```suggestion
    description: 'Database name'
```

File: core/action.yml [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    required: false
+    default: ""
+  migration_envs:
+    description: 'List of environment variables that will be set during migrations running'
```
Comment (NikolayS) [inline core/action.yml:44]:
```suggestion
    description: 'List of environment variables that will be set before testing'
```

File: core/action.yml [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    required: false
+  download_artifacts:
+    description: 'Option to download artifacts'
```
Comment (NikolayS) [inline core/action.yml:47]:
```suggestion
    description: 'Store artifacts and allow downloading them'
```

File: core/action.yml [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    required: false
+    default: "false"
+
+  observation_interval:
+    description: 'Observation interval'
```
Comment (NikolayS) [inline core/action.yml:52]:
```suggestion
    description: 'Observation interval, seconds (default: 10)'
```

File: core/action.yml [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+    required: false
+    default: "10"
+  max_lock_duration:
+    description: 'Max lock duration'
```
Comment (NikolayS) [inline core/action.yml:56]:
```suggestion
    description: 'Exclusive lock duration threshold, seconds (default: 10)'
```

File: core/action.yml [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+    required: false
+    default: "10"
+  max_duration:
+    description: 'Max duration'
```
Comment (NikolayS) [inline core/action.yml:60]:
```suggestion
    description: 'DB migration duration threshold, seconds (default: 3600)'
```

File: core/action.yml [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+    required: false
+    default: "3600"
+
+
+outputs:
+  response:
+    description: 'The result of CI checks'
```
Comment (NikolayS) [inline core/action.yml:67]:
```suggestion
    description: 'DB migration testing results'
```

**Review by NikolayS — APPROVED** [review]
+1 +1 +1 @agneum 

can you please include an example of outputs (screenshot(s))?

Small correction of all texts – included in comments (common issue – let's say "testing" rather than "checking")

**Comment by NikolayS** [comment]
> can you please include an example of outputs (screenshot(s))?

ignore it – I see the links, I guess we need to start developing the formatted output at this stage already.

File: action.yml (hunk 1 above)
Comment (agneum) [reply action.yml:49]: Is there any way to distinguish them?

**Pull request merged**